packages/flutter/lib/src/material/app.dart

```diff
@@ -297,6 +312,29 @@ class _MaterialAppState extends State<MaterialApp> {
     _heroController = new HeroController(createRectTween: _createRectTween);
   }
 
+  @override
+  void didUpdateWidget(MaterialApp old) {
+    super.didUpdateWidget(old);
+    /* TBD
+    final LocalizationsDelegate delegate = widget.localizationsDelegate;
+    final LocalizationsDelegate oldDelegate = old.localizationsDelegate;
+    if ((delegate == null && oldDelegate != null)
+        || (delegate != null && delegate.shouldReload(oldDelegate)))
+      _localizationsDelegate = _createLocalizationsDelegate();
+    */
+  }
+
+  // Combine the Localizations for Material with the ones contributed
+  // by the localizationsDelegates parameter, if any.
+  Iterable<LocalizationsDelegate<dynamic>> _createLocalizationsDelegates() {
+    final List<LocalizationsDelegate<dynamic>> delegates = <LocalizationsDelegate<dynamic>>[
+      const _MaterialLocalizationsDelegate(),
+    ];
+    if (widget.localizationsDelegates != null)
+      delegates.addAll(widget.localizationsDelegates);
+    return delegates;
+  }
```

Comment:
since you're using Iterables you can write this function as:

```dart
Iterable<LocalizationsDelegate<dynamic>> _createLocalizationsDelegates() sync* {
  yield const _MaterialLocalizationsDelegate();
  if (widget.localizationsDelegates != null)
    yield* widget.localizationsDelegates;
}
```